File: full_case_study_solns/_f_planar_vtol/python_old/hw7/VTOLParamHW7.py
# VTOL Parameter File
import numpy as np
# import control as cnt
import sys
sys.path.append('..')  # add parent directory
import VTOLParam as P

# select desired closed loop char eq
Delta_cl_d = np.poly([-.3,-.2])

# PD gains
kp_h = Delta_cl_d[2]*(P.mc+2.0*P.mr)
kd_h = Delta_cl_d[1]*(P.mc+2.0*P.mr)


print('kp_h: ', kp_h)
print('kd_h: ', kd_h)

File: full_case_study_solns/_f_planar_vtol/matlab_old/hw7/VTOLParamHW7.m
% inverted ballbeam - parameter file for hw8
addpath ./.. % adds the parent directory to the path
VTOLParam % general ballbeam parameters

% select desired closed loop char eq
Delta_cl_d = poly([-.3,-.2]);

% PD gains
P.kp_h = Delta_cl_d(3)*(P.mc+2*P.mr);
P.kd_h = Delta_cl_d(2)*(P.mc+2*P.mr);

fprintf('\t kp_z: %f\n', P.kp_h)
fprintf('\t kd_z: %f\n', P.kd_h)
